Spigot-API-Patches/0157-Add-spectator-target-events.patch

```diff
@@ -0,0 +1,121 @@
+From 6399e113a3a6c2f4f3f11f5363b0087aeae079f1 Mon Sep 17 00:00:00 2001
+From: Caleb Bassham <caleb.bassham@gmail.com>
+Date: Fri, 28 Sep 2018 02:30:56 -0500
+Subject: [PATCH] Add spectator target events
+
+- PlayerStartSpectatingEntityEvent
+- PlayerStopSpectatingEntityEvent
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/player/PlayerStartSpectatingEntityEvent.java b/src/main/java/com/destroystokyo/paper/event/player/PlayerStartSpectatingEntityEvent.java
+new file mode 100644
+index 00000000..fd571525
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/player/PlayerStartSpectatingEntityEvent.java
+@@ -0,0 +1,52 @@
++package com.destroystokyo.paper.event.player;
++
++import org.bukkit.entity.Entity;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++
++/**
++ * Called when a player starts spectating an entity in spectator mode.
++ */
++public class PlayerStartSpectatingEntityEvent extends PlayerEvent implements Cancellable {
++
++    private static final HandlerList handlers = new HandlerList();
++    private boolean cancelled;
++    private final Entity currentSpectatorTarget;
++    private final Entity newSpectatorTarget;
++
++    public PlayerStartSpectatingEntityEvent(Player player, Entity currentSpectatorTarget, Entity newSpectatorTarget) {
++        super(player);
++        this.currentSpectatorTarget = currentSpectatorTarget;
++        this.newSpectatorTarget = newSpectatorTarget;
++    }
++
++    public Entity getCurrentSpectatorTarget() {
```

Comment:
Even basic docs like "The entity being spectated" often lend clarity, please add them API side.